qiskit/qobj/qasm_qobj.py

```diff
@@ -313,70 +315,127 @@ def __eq__(self, other):
+
+
+class QasmExperimentCalibrations:
+    """A container for any calibrations data. The gates attribute contains a list of
+    GateCalibrations.
     """
 
-    def __init__(self, **kwargs):
-        """Instantiate a new Qobj dict field object.
+    def __init__(self, gates):
+        """
+        Initialize a container for calibrations.
 
         Args:
-            kwargs: arbitrary keyword arguments that can be accessed as
-                attributes of the object.
+            gates (list(GateCalibration))
         """
-        self.__dict__.update(kwargs)
+        self.gates = gates
 
     def to_dict(self):
-        """Return a dictionary format representation of the QASM Qobj.
+        """Return a dictionary format representation of the calibrations.
 
         Returns:
-            dict: The dictionary form of the QobjHeader.
+            dict: The dictionary form of the GateCalibration.
 
         """
-        return self.__dict__
+        out_dict = copy.copy(self.__dict__)
+        out_dict['gates'] = [x.to_dict() for x in self.gates]
+        return out_dict
 
     @classmethod
     def from_dict(cls, data):
-        """Create a new QobjHeader object from a dictionary.
+        """Create a new GateCalibration object from a dictionary.
 
         Args:
-            data (dict): A dictionary representing the QobjHeader to create. It
-                will be in the same format as output by :func:`to_dict`.
+            data (dict): A dictionary representing the QasmExperimentCalibrations to
+                         create. It will be in the same format as output by :func:`to_dict`.
 
         Returns:
-            QobjDictFieldr: The QobjDictField from the input dictionary.
+            QasmExperimentCalibrations: The QasmExperimentCalibrations from the input dictionary.
         """
-
+        gates = data.pop('gates')
+        data['gates'] = [GateCalibration.from_dict(x) for x in gates]
         return cls(**data)
 
-    def __eq__(self, other):
-        if isinstance(other, self.__class__):
-            if self.__dict__ == other.__dict__:
-                return True
-        return False
 
+class GateCalibration:
+    """Each calibration specifies a unique gate by name, qubits and params, and
+    contains the Pulse instructions to implement it."""
 
-class QasmQobjExperimentHeader(QobjDictField):
-    """A header for a single QASM experiment in the qobj."""
-    pass
+    def __init__(self, name, qubits, params, instructions):
+        """
+        Initialize a single gate calibration. Instructions may reference waveforms which should be
+        made available in the pulse_library.
 
+        Args:
+            name (str): Gate name.
+            qubits (list(int)): Qubits the gate applies to.
+            params (list(complex)): Gate parameter values, if any.
+            instructions (list(PulseQobjInstruction)): The Pulse instructions which implement
+                                                       the gate.
```

Comment:
I think the indentation of the docstring need not be aligned this way. I'd prefer something similar to L430-431. I am okay with either though.


```
            data (dict): A dictionary representing the GateCalibration to create. It
                will be in the same format as output by :func:`to_dict`.
```